app/main.py

```diff
@@ -37,7 +36,6 @@ def __getitem__(self, key: Any) -> Any:
             index += 1
             if index == len(self.hash_table):
                 index = 0
-            if index == len(self.hash_table):
                 return self.hash_table[index][2]
```

Comment:
this line is also unnecessary because when the index becomes 0 you also need to compare the element under that index with a key and hash and move forward if necessary. In your code, you always return an element under the 0 index without checking it